src/main/java/mat/client/shared/CQLLibraryResultTable.java

```diff
@@ -62,41 +64,75 @@ public Boolean getValue(CQLLibraryDataSetObject object) {
 			public SafeHtml getValue(CQLLibraryDataSetObject object) {
 				return getCQLLibraryNameColumnToolTip(object);
 			}
-		};
-//		Double Click event on Grid row to navigate to Composer Tab.
-		if(MatContext.get().getMatOnFHIR().getFlagOn()) {
-			table.addDomHandler(new DoubleClickHandler() {
-				@Override
-				public void onDoubleClick(DoubleClickEvent event) {
-					for (CQLLibraryDataSetObject object : selectionModel.getSelectedSet()) {
-						SelectionEvent.fire(fireEvent, object);
-					}
+			/*
+				Single Click to select row and enable checkbox
+				Double Click to navigate to CQL Composer tab
+			*/
+			@Override
+			public void onBrowserEvent(Cell.Context context, Element elem, CQLLibraryDataSetObject object, NativeEvent event) {
+				if (lastClick < System.currentTimeMillis() - 500)
```

Comment:
500 is a magic number. should be either a constant or come from a config file

Reply:
Noted